pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -0,0 +1,213 @@
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.keithley.keithley2600 import Keithley2600
+from pymeasure.instruments.validators import strict_discrete_set, truncated_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Keithley2602B(Keithley2600):
+    """Represents the Keithley 2602B SourceMeter. This class adds digital I/O
+    functionality to the Keithley2600 series driver. Note that this driver can
+    be used to control any of the other 2600 series models that support digital
+    I/O (2601B, 2611B, 2612B, 2635B, 2636B).
+    """
+
+    number_of_pins = 14
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, includeSCPI=False, **kwargs)
```

Comment:
The `includeSCPI` flag toggles the inclusion of standard SCPI commands (e.g. `*IDN?`) included in PyMeasure's base `Instrument` class. This is false because this SMU uses commands written in Lua, so the standard SCPI commands are not compatible - I am not sure why they didn't set this in the original driver, hopefully will get some feedback on the PyMeasure PR.